src/Common/Infrastructure/BackendResponse.cs

```diff
@@ -12,5 +15,48 @@ public BackendResponse(string content, string contentType, bool cameFromCache)
             ContentType = contentType;
             CameFromCache = cameFromCache;
         }
+
+        public void UpdateNextPageUrlWithQueryParameters(NonPagedQueryCollection requestQuery, string nextPageUrlOption)
+        {
+            if (string.IsNullOrWhiteSpace(nextPageUrlOption) || requestQuery.IsEmpty)
+                return;
+
+            const string numberPatternPlaceholder = "__NUMBER_PATTERN_PLACEHOLDER_THAT_WONT_BE_REGEX_ESCAPED__";
+            var nextPageUrlOptionWithPlaceholders = string.Format(nextPageUrlOption, numberPatternPlaceholder, numberPatternPlaceholder);
+            var nextPageUrlValePattern = Regex
+                .Escape(nextPageUrlOptionWithPlaceholders)
+                .Replace(numberPatternPlaceholder, "\\d+");
+
+            var parameters = requestQuery
+                .Aggregate("", (result, filterQueryParameter) => $"{result}&{filterQueryParameter.Key}={filterQueryParameter.Value}");
+
+            var isXml = ContentType.ToLower().Contains("xml");
```

Comment:
Can we add a new class `ContentType` like we did with `AcceptType`? https://github.com/Informatievlaanderen/public-api/blob/134493d8f13e8e8618e87ffa614f23999337414b/src/Common/Infrastructure/AcceptType.cs

Then we can get rid of the stringy contenttype in here as well

Reply:
ContentType is just response property, too much overhead to create type